pkg/controller/workloadspread/workloadspread_controller.go

```diff
@@ -327,16 +328,49 @@ func groupPod(ws *appsv1alpha1.WorkloadSpread, pods []*corev1.Pod) map[string][]
 	for i := range pods {
 		injectWS := getInjectWorkloadSpreadFromPod(pods[i])
 		if injectWS == nil || injectWS.Name != ws.Name {
-			continue
-		}
-		if _, exist := podMap[injectWS.Subset]; exist {
+			// process the pods that were created before workloadSpread
+			if matchedSubset, err := r.getSuitableSubsetForOldPod(ws, pods[i]); err != nil {
+				klog.Errorf("failed to get a suitable subset for the old pod (%s/%s) which was created before workloadSpread (%s/%s), err: %v",
+					pods[i].Namespace, pods[i].Name, ws.Namespace, ws.Name, err)
+			} else if matchedSubset != nil {
+				podMap[matchedSubset.Name] = append(podMap[matchedSubset.Name], pods[i])
+			}
+		} else if _, exist := podMap[injectWS.Subset]; exist {
 			podMap[injectWS.Subset] = append(podMap[injectWS.Subset], pods[i])
 		}
 	}
 
 	return podMap
 }
 
+// getSuitableSubsetForOldPod returns an suitable subset for the pod that was created before workloadSpread.
+// If there is no suitable subset for the pod, this function will return a empty pointer.
+func (r *ReconcileWorkloadSpread) getSuitableSubsetForOldPod(ws *appsv1alpha1.WorkloadSpread, pod *corev1.Pod) (*appsv1alpha1.WorkloadSpreadSubset, error) {
+	var err error
+	var node *corev1.Node
+
+	if len(pod.Spec.NodeName) > 0 {
+		node = &corev1.Node{}
+		err = r.Get(context.TODO(), types.NamespacedName{Name: pod.Spec.NodeName}, node)
+	}
+
+	if err != nil && !errors.IsNotFound(err) {
```

Comment:
consider move the subset iteration out of the if clause to make the main logic clear. 
```
  if len(pod.Spec.NodeName) == 0 {
      return nil, nil
   }
  if err != nil {
        if errors.IsNotFound(err) {
            return nil, nil 
        }
        return nil, err
   }
  for i := range ws.Spec.Subsets {
  }

```

Reply:
done

Reply:
done